### PR TITLE
fix nomad job reload after resuming autoscaling

### PR DESCRIPTION
**Pull request by jvineet**
This has potential fix for autoscale reverting docker image tag/version problem. Currently running on test, appears to be working, though there is not really a controlled way to test it.

### CODE REVIEW

File: scaler.go [hunk 1]
```diff
@@ -63,6 +63,22 @@ func (scaler *Scaler) resumeScallingJob(w http.ResponseWriter, r *http.Request)
 	defer mutex.Unlock()
 	delete(scaler.jobMap, mapID)
 
+	jobMapMutex.Lock()
+	jobMapScaleMutex.Lock()
+
+	log.Debug("Updating job config for ", jobID)
```
Comment (dbachelder) [inline scaler.go]: is this the right log message? maybe something about clearing or deleting job config? or job map?

File: scaler.go [hunk 2]
```diff
@@ -63,6 +63,22 @@ func (scaler *Scaler) resumeScallingJob(w http.ResponseWriter, r *http.Request)
...
+	delete(jobMap, jobID)
```
Comment (dbachelder) [inline scaler.go]: why isn't this `scaler.jobMap` like it is on line 64?

File: scaler.go [hunk 3]
```diff
@@ -63,6 +63,22 @@ func (scaler *Scaler) resumeScallingJob(w http.ResponseWriter, r *http.Request)
...
+	delete(jobMapScale, jobID)
```
Comment (dbachelder) [inline scaler.go]: I don't see any use of this jobMapScale besides the code you added... that seems weird?

Comment (jvineet) [reply scaler.go]: yeah, I am also very confused how that is being used. It gets scaling jobs added in the `checkMeta(jobMap map[string]*api.Job)` function. But then I couldn't see if this was being used to anything. I just deleted from that struct here as I saw that is how the `getJobs()` function is updating all job configuration, to keep consistent.

File: scaler.go (hunk 2 above)
Comment (jvineet) [reply scaler.go]: scaler.jobMap is a different struct. This struct doesn't seem to hold complete nomad job config information. The jobMap struct is defined in `main.go`

#in scaler.go
```
type Scaler struct {
	jobMap map[string]structs.JobStruct
}
```

#in structs.go
```
type JobStruct struct {
	JobName           string
	Region            string
	ScaleMin          int
	ScaleMax          int
	ScaleCountUp      int
	ScaleCountDown    int
	ScaleCooldown     time.Time
	ScaleCooldownUp   time.Time
	ScaleCooldownDown time.Time
	LastRun           time.Time
	Count             int
	Group             int
	NoGo              bool
	EndValue          int
	GroupName         string
	TaskName          string
}

```

The jobMap struct in main.go, is defined like this
```
map[string]*nomad.Job.  (nomad_job_id -> nomad.Job struct pointer mapping)
```
I think nomad.Job struct is defined in the nomad client, and the Nomad api get function returns a pointer to this struct.

File: scaler.go (hunk 1 above)
Comment (jvineet) [reply scaler.go]: I am deleting the old Nomad Job struct pointer here from the JobMap, and then adding the updated one after the api call. So I thought of that as update. I am doing a delete and add as `getJobs()`, which updates all job configuration, appears to be doing something similar. 

Maybe `Refreshing job config`

**Pull request merged**